### ISSUE TITLE
navigation history: hyperlinking from inside EPUB XHTML documents

### ISSUE THREAD

**Issue by danielweck** [edrlab/thorium-reader#1953]
when following a TOC link from within an EPUB content document (or for that matter, any hyperlink such as a footnote link), the "previous" jump goes to wherever the previous major navigation point was (for example a few pages back when the user landed on the cover image, and then flipped pages to reach the inline authored TOC) .... this should be overridden by the location of the hyperlink itself! Solution: before activating / following the hyperlink, make sure to inject a navigation milestone into the history at the reading location where the hyperlink is located (i.e. getCurrentReadingLocation()).

**Comment by danielweck** [comment]
Fixed via https://github.com/readium/r2-navigator-js/commit/9b3ac4045fdc09da06345f6420605d9a41047650
